Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/KeyedHashFileViewModel.cs

```diff
@@ -89,54 +89,70 @@ protected override void InitializeModel()
+            {
+                DisplayName = Resources.Activity_KeyedHashFile_Property_Key_SwitchName,
+                Handler = UseKeyHandler
+            };
+
+            var useKeySecureStringMenuAction = new MenuAction
+            {
+                DisplayName = Resources.Activity_KeyedHashFile_Property_KeySecureString_SwitchName,
+                Handler = UseKeySecureStringHandler
+            };
+
+            _useKeySecureString = KeySecureString.Value != null;
+
+            Key.AddMenuAction(useKeySecureStringMenuAction);
+            KeySecureString.AddMenuAction(useKeyMenuAction);
 
             Result.IsPrincipal = false;
             Result.OrderIndex = propertyOrderIndex++;
 
             ContinueOnError.IsPrincipal = false;
             ContinueOnError.OrderIndex = propertyOrderIndex++;
-            ContinueOnError.Widget = new DefaultWidget { Type = ViewModelWidgetType.NullableBoolean };
+            ContinueOnError.Widget = new DefaultWidget { Type = "NullableBoolean", Metadata = new Dictionary<string, string>() };
             ContinueOnError.Value = false;
-
-            MenuActionsBuilder<KeyInputMode>.WithValueProperty(KeyInputModeSwitch)
-                .AddMenuProperty(Key, KeyInputMode.Key)
-                .AddMenuProperty(KeySecureString, KeyInputMode.SecureKey)
-                .BuildAndInsertMenuActions();
         }
 
-        /// <inheritdoc/>
-        protected override void InitializeRules()
+        private Task UseKeyHandler(MenuAction action)
         {
-            base.InitializeRules();
-            Rule(nameof(KeyInputModeSwitch), KeyInputModeChanged_Action);
+            Key.IsVisible = true;
+            Key.IsRequired = true;
+            KeySecureString.IsRequired = false;
+            KeySecureString.IsVisible = false;
+            KeySecureString.Value = null;
+            _useKeySecureString = false;
+            return Task.CompletedTask;
         }
 
-        /// <inheritdoc/>
-        protected override void ManualRegisterDependencies()
+        private Task UseKeySecureStringHandler(MenuAction action)
         {
-            base.ManualRegisterDependencies();
-            RegisterDependency(KeyInputModeSwitch, nameof(KeyInputModeSwitch.Value), nameof(KeyInputModeSwitch));
+            Key.IsVisible = false;
+            Key.IsRequired = false;
+            KeySecureString.IsRequired = true;
+            KeySecureString.IsVisible = true;
+            Key.Value = null;
+            _useKeySecureString = true;
+            return Task.CompletedTask;
         }
 
-        /// <summary>
-        /// Key input Mode has changed. Set controls visibility based on selection
-        /// </summary>
-        private void KeyInputModeChanged_Action()
+        /// <inheritdoc/>
+        protected override void InitializeRules()
         {
-            switch (KeyInputModeSwitch.Value)
+            base.InitializeRules();
+            Rule("VisibleKey", () =>
```

Comment:
Can we use `nameof(PropertyName)` instead of hardcoding strings here ?

Reply:
It's the name of the rule defined there, I don't think there are variables which can be put there for naming a rule.

Reply:
How this should be handled is as follows (or at least how it's handled in Office and other activities).

* Each property should have its own rule (e.g. for key, for key secure string, etc)
* Each rule should have its own handler method
* In the handler, you can call the same private method or methods to update other properties. The idea is that any change on the value of property A that affects properties B, C, and D should call "UpdateStateOfB", "UpdateStateOfC" etc